resource-timing/image-sequence-of-events.html

```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Test the sequence of events when reporting image timing.</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+  function test_image_sequence(src, event, t) {
+    const image = document.createElement('img');
+    const sequence = [];
+    image.src = src;
+    const absoluteURL = new URL(src, location.href).toString();
+    document.body.appendChild(image);
+    t.add_cleanup(() => image.remove());
+    return new Promise(resolve => image.addEventListener(event, t.step_func(() => {
```

Comment:
I suspect this would be racy on WebKit, because last time I looked, they're not following the spec and setting `img.src` there doesn't wait a microtask before loading the resource. It'd be better to set the src inside the promise and after setting the event listener.

Reply:
Fixed